# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363804999 25200
#      Wed Mar 20 11:43:19 2013 -0700
# Node ID b73b5ab6c277a65bf2e51ed9cd54bcb08cdde068
# Parent  c34f8bb1ebece868563f3c89ffca86e31de9a02e
Remove extra line printed at end of dbtype (off-by-1 error).

* libinterp/interpfcn/debug.cc: Don't print line number at start of line
until fs.get () has checked that this is not EOF.

diff --git a/libinterp/interpfcn/debug.cc b/libinterp/interpfcn/debug.cc
--- a/libinterp/interpfcn/debug.cc
+++ b/libinterp/interpfcn/debug.cc
@@ -759,32 +759,36 @@ do_dbtype (std::ostream& os, const std::
   if (! ff.empty ())
     {
       std::ifstream fs (ff.c_str (), std::ios::in);
 
       if (fs)
         {
           char ch;
           int line = 1;
-
-          if (line >= start && line <= end)
-            os << line << "\t";
+          bool isnewline = true;
 
-          while (fs.get (ch))
+          // FIXME: Why not use line-oriented input here [getline()]?
+          while (fs.get (ch) && line <= end)
             {
-              if (line >= start && line <= end)
+              if (isnewline && line >= start)
+                {
+                  os << line << "\t";
+                  isnewline = false;
+                }
+
+              if (line >= start)
                 {
                   os << ch;
                 }
 
               if (ch == '\n')
                 {
                   line++;
-                  if (line >= start && line <= end)
-                    os << line << "\t";
+                  isnewline = true;
                 }
             }
         }
       else
         os << "dbtype: unable to open '" << ff << "' for reading!\n";
     }
   else
     os << "dbtype: unknown function " << name << "\n";
